### ISSUE TITLE
Metadata is not being sync’d, maybe related to --s3-decompress

### ISSUE THREAD

**Issue by manzsolutions-lpr** [rclone/rclone#6546]
#### What is the problem you are having with rclone?

Metadata is no longer synchronised with 1.60.0.

This works using the `-M/--metadata` parameter in 1.59.2 (see the log in the Gist below). This may be related to the `--s3-decompress` parameter which now also needs to be used and apparently had a bug regarding [metadata](https://github.com/rclone/rclone/issues/6434).

#### What is your rclone version (output from `rclone version`)

```
rclone v1.60.0
- os/version: debian 11.5 (64 bit)
- os/kernel: 5.18.0-0.deb11.4-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.19.2
- go/linking: static
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Debian 11, 64 Bit

#### Which cloud storage system are you using? (e.g. Google Drive)

Netapp StorageGrid (S3 compatible)

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

```
rclone sync -M --s3-decompress test-bucket:test-bucket-et-created/ test-bucket-et-created_post_restore
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

https://gist.github.com/manzsolutions-lpr/a8bcad93cbf28fab531db0a83f3a1c32

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
Can you try [the latest beta](https://beta.rclone.org/) which should fix the decompress problem.

If this doesn't fix your problem then please post another log.

I plan to put the fix for the decompress problem in v1.60.1

**Comment by manzsolutions-lpr** [comment]
Yes, the beta seems to work again, thank you! Looking forward to 1.60.1 then.

**Comment by ncw** [comment]
> Yes, the beta seems to work again, thank you! Looking forward to 1.60.1 then.

Great!

You can try a beta of v1.60.1 here: https://beta.rclone.org/branch/v1.60-stable/ 

> //edit: But not if `--s3-decompress` is specified explicitly. Is this intentional?
> 
> //edit 2: Providing a log but I’m not really sure if the parameter should be even used in our use case or what its actual purpose is.
> 
> https://gist.github.com/manzsolutions-lpr/b911b4b87d3cd2d9b84856cdc7cd6a5c

The log was useful - thanks.

Providing `--s3-decompress` erases the size of compressed objects as rclone wont know how big they are when decompressed. This means rclone falls back to `RCat` to copy file streams.

However there is a bug in the `RCat` code which isn't preserving the metadata.

I've made a fix for this here - if you copy files even with --s3-decompress it should preserve their metadata. Please give it a go!

[v1.61.0-beta.6544.771bdc357.fix-6546-rcat-metadata](https://beta.rclone.org/branch/fix-6546-rcat-metadata/v1.61.0-beta.6544.771bdc357.fix-6546-rcat-metadata/) on branch [fix-6546-rcat-metadata](https://github.com/rclone/rclone/tree/fix-6546-rcat-metadata) (uploaded in 15-30 mins)

Unfortunately this fix is too big for the point release I think - I had to change quite a lot of code. I don't think you'll need `--s3-decompress` in normal operation.

**Comment by ncw** [comment]
I've merged the fix for Rcat to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.61. I'm not going to put in it 1.60.1 as it is too disruptive for something which very few people will notice.
